doc/scapy/development.rst

```diff
@@ -300,10 +300,29 @@ publish the release::
 Release Candidates (RC) could also be done. For example, the first RC will be
 tagged v2.4.3rc1 and the message ``2.4.3 Release Candidate #1``.
 
-Prior to uploading the release to PyPi, the ``author_email`` in ``setup.py``
-must be changed to the address of the maintainer performing the release. The
-following commands can then be used::
+Prior to uploading the release to PyPi, the mail address of the maintainer
+performing the release must be added next to his name in ``pyproject.toml``.
+See `this <https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#authors-maintainers>`_ for details.
 
- python3 setup.py sdist
- twine check dist/scapy-2.4.3.tar.gz
- twine upload dist/scapy-2.4.3.tar.gz
+The following commands can then be used::
+
+ pip install --upgrade build
+ python -m build
+ twine check dist/*
+ twine upload dist/*
+
+.. warning::
+   Make sure that you don't have left-overs in your ``dist/`` folder ! There should only be the source and the wheel for the package.
+   Also check that the wheel ends in ``*-py3-none-any.whl`` !
+
+
+Packaging Scapy
+===============
+
+When packaging Scapy, you should build the source while setting the ``SCAPY_VERSION`` variable, in order to make sure that the version remains consistent.
+
+.. code:: bash
+
+   $ SCAPY_VERSION=2.5.0 python3 -m build
+   ...
+   Successfully built scapy-2.5.0.tar.gz and scapy-2.5.0-py3-none-any.whl
```

Comment:
I think the "scapy/VERSION" approach where it's added after unpacking the tarball still works so both Debian and Gentoo can keep using their build scripts but `SCAPY_VERSION` (or the git archive tag used by Arch and Apline) seems to be more reliable.

@thesamesam just a heads-up. `echo ${PV} > ${PN}/VERSION || die` in https://gitweb.gentoo.org/repo/gentoo.git/tree/net-analyzer/scapy/scapy-2.5.0.ebuild#n48 can be removed and replaced with `SCAPY_VERSION` when the next `scapy` version is released. It should be possible to remove it now and rely on the git archive tag though. Also the tests should be excluded from the sdist and wheel so it should be possible to remove https://gitweb.gentoo.org/repo/gentoo.git/tree/net-analyzer/scapy/files/scapy-2.5.0-no-install-tests.patch (when `scapy` with all these changes is released).

FWIW on an unrelated note looking at https://gitweb.gentoo.org/repo/gentoo.git/tree/net-analyzer/scapy/scapy-2.5.0.ebuild I think instead of removing the tests it should be possible to just skip them by passing `-K ...` to `UTscapy`. For example, `-K netaccess` should replace https://gitweb.gentoo.org/repo/gentoo.git/tree/net-analyzer/scapy/scapy-2.5.0.ebuild#n54.

Reply:
A note on that: the idea of [`test/run_tests`](https://github.com/secdev/scapy/blob/master/test/run_tests) is to have a script called by package maintainers that runs tests that don't require external dependencies, and not having each maintainer chose which tests to run. Might be worth adding some doc.

Reply:
@gpotter2 I think `run_tests` uses `tox` to create a new test virtual environment and some distros can't do that so they have to run `UTscapy` directlly.

Reply:
FWIW it should be possible to run `./run_tests` in environments where `tox` isn't allowed to install dependencies with pip. To get the tests to run using the Fedora infrastructure I put the test dependencies in "BuildRequires", installed https://pypi.org/project/tox-current-env/ (Fedora ships it as a usual package) and added a sed command adding `--current-env` to `.config/ci/test.sh` to prevent `tox` from installing stuff with `pip`: https://github.com/evverx/scapy/pull/2/files#diff-0673ea92d72beb0d23ba537e1e83854385c56c702ef9abbed890b4a31ed4ece0R20

It's kludgy but it works :-) Running `UTscapy` directly is kind of hard because it already expects for example the openssl config to be generated and it's just easier to let `.config/ci/test.sh` do it.

Anyway thanks for updating the documentation! I think this thread can probably be resolved.

Reply:
Thanks a bunch for the heads up @evverx!